### DISCUSSION TITLE
KeyErrror: ‘Unable to open object (bad object header version number’

### DISCUSSION THREAD

**Discussion by tserpling** [talmolab/sleap#918] [Help!]
Hello, 

I was labeling track identities for my project on an HP laptop when it suddenly froze as I clicked an instance to label. I did a hard reboot by holding down the power button of the computer. When I restarted the laptop and opened SLEAP again, I tried to upload the project, but I got this message: 

KeyErrror: ‘Unable to open object (bad object header version number’.

After a basic search on the internet, I’m under the assumption that my file is corrupted. Is there a way to fix the file? Or how can I restore it to an uncorrupted version (I saved my project frequently; every 2-5 frames or so before it froze). 

Thanks, 
Tara

**Comment by roomrys** [comment]
Hi @tserpling,

Oh man, this doesn't sound good. Since the computer crashed, the hdf5 did not close properly leading to the corrupted file. 

Do you have any back-ups saved under a different filename or file versions via something like [Windows backup](https://support.microsoft.com/en-us/windows/recover-lost-or-deleted-files-7bf065bf-f1ea-0a78-c1cf-7dcf51cc8bfc#:~:text=Right%2Dclick%20the%20file%20or%20folder%2C%20and%20then%20select%20Restore,if%20both%20types%20are%20available.)? [This thread](https://groups.google.com/g/h5py/c/TM0W-0c90TY) suggestions using h5clear, which might be worth a shot if you do not have another version of your file.

Fingers crossed that you are able to recover a previous version of the file. It might be worth it to set-up a backup manager and/or Save As throughout the lifetime of your project - advice given a little too late :cry:.

Sorry to hear about this, best of luck, and keep us updated (with news and/or if we can assist). 

Thanks,
Liezl

**Reply by IsabelleSajonia** [reply]
Hi I had the same error when my PC crashed. Can I use a predictions slp file to train (remove predictions and label new frames)? Would this continue updating my training and validation set so I can evaluate the model?

**Reply by roomrys** [reply]
Hi @IsabelleSajonia,

You can use any slp to train as long as it has a few user labels. Each round of training creates a new/independent training and validation split. However, starting from the predictions.slp would essentially be starting starting over (with help from a model to get started). Thus, the frames under the user labels (which are split to create the training/validation set) would not be that same set of user labels that was used before.

Liezl
